aspnetcore/mvc/controllers/filters.md

```diff
@@ -1,124 +1,167 @@
 ---
 title: Filters in ASP.NET Core
 author: ardalis
-description: Learn how filters work and how to use them in ASP.NET Core MVC.
+description: Learn how filters work and how to use them in ASP.NET Core.
 ms.author: riande
 ms.custom: mvc
-ms.date: 02/08/2019
+ms.date: 5/08/2019
 uid: mvc/controllers/filters
 ---
 # Filters in ASP.NET Core
 
+https://localhost:44358/Sample/HeaderWithFactory
+
 By [Rick Anderson](https://twitter.com/RickAndMSFT), [Tom Dykstra](https://github.com/tdykstra/), and [Steve Smith](https://ardalis.com/)
 
-*Filters* in ASP.NET Core MVC allow you to run code before or after specific stages in the request processing pipeline.
+*Filters* in ASP.NET Core allow code to be run before or after specific stages in the request processing pipeline.
 
 Built-in filters handle tasks such as:
 
 * Authorization (preventing access to resources a user isn't authorized for).
 * Ensuring that all requests use HTTPS.
-* Response caching (short-circuiting the request pipeline to return a cached response). 
+* Response caching (short-circuiting the request pipeline to return a cached response).
+
+Custom filters can be created to handle cross-cutting concerns. Examples of cross-cutting concerns include error handling, caching, configuration, authorization, and logging.  Filters avoid duplicating code. For example, an error handling exception filter could consolidate error handling.
 
-Custom filters can be created to handle cross-cutting concerns. Filters can avoid duplicating code across actions. For example, an error handling exception filter could consolidate error handling.
+This document applies to Razor Pages, API controllers, and controllers with views.
 
 [View or download sample from GitHub](https://github.com/aspnet/AspNetCore.Docs/tree/master/aspnetcore/mvc/controllers/filters/sample).
 
 ## How filters work
 
-Filters run within the *MVC action invocation pipeline*, sometimes referred to as the *filter pipeline*.  The filter pipeline runs after MVC selects the action to execute.
+Filters run within the *ASP.NET Core action invocation pipeline*, sometimes referred to as the *filter pipeline*.  The filter pipeline runs after ASP.NET Core selects the action to execute.
```

Comment:
I guess MVC has been dropped here because of RP, but "action" seems to be more of an MVC term so I wonder if this non-MVC concept of an "action" might be confusing.